node/clients/search.ts

```diff
@@ -80,17 +83,26 @@ export class Search extends AppClient {
     )
   }
 
-  public product = (slug: string) =>
+  public product = (slug: string, vtexSegment?: string | undefined,  salesChannel?: string | number) =>
```

Comment:
```suggestion
  public product = (slug: string, vtexSegment?: string,  salesChannel?: string | number) =>
```
the `?` already defines that this value can be `undefined`